docs/networking.md

```diff
@@ -0,0 +1,134 @@
+# Networking
+
+By default, linuxkit recognizes network devices but does not configure them.
+You must provide either static configuration or use dhcpcd (onboot or as a service)
+to get an IP.
+
+The normal (and simplest) use-case is just including dhcpcd into the onboot section,
+this will cause dhcpcd to startup *once* and assign an IP:
+```
+onboot:
+  - name: dhcpcd
+    image: linuxkit/dhcpcd:<hash>
+    command: ["/sbin/dhcpcd", "--nobackground", "-f", "/dhcpcd.conf", "-1"]
+```
+
+The alternative method is to make it a service. This ensures dhcpcd catches any updates,
+but may cause issues with services that expect an IP address on startup,
+since service launch order is *not* guaranteed.
+
+Similarly, using WiFi requires wpa_supplicant in addition to dhcpd.
+
+Finally, if you require any specialized firmware or device drivers for your network card,
+you may need to load it.
+
+While the common setup of LinuxKit doesn't usually bring any network connectivity issues,
+they still may occur. Moreover, LinuxKit may be used on the real hardware,
+and in that case the network setup could be a major part of the getting the image run.
+
+## General Steps
+
+There is a standard checklist one have to follow to set up the networking:
+
+1. Drivers are loaded by kernel (otherwise, may need a [custom kernel](kernels.md));
+2. Firmware is loaded (otherwise, need to put firmware blobs into the image);
+3. Network interfaces are visible in the system (otherwise, need to revisit items 1 and 2);
+4. Wi-Fi: wpa_supplicant is running (otherwise, need to include it into services section);
+5. DHCP: dhcpcd service is loaded.
+6. May need to set up static IP addressed and routes, but it's out of scope of this document.
+
+Usually, `lspci` provides the information about the detected devices, and
+`dmesg` prints the information whether the drivers for those devices
+are loaded or not (usually if nothing mentioned in the log, then no driver is loaded).
+
+### Wi-Fi Notes
```

Comment:
I would make this a heading one level up, ie `##` and have the wifi specific bits be under this heading